### PR TITLE
Feat/fastapi deployment

### PR DESCRIPTION
**Pull request by degrootdev**
In the readme is explained how to use the api. I only tested it with the baseline model, so I am not sure if it will already work with the u-net model. Let me know if you have any feedback

### CODE REVIEW

File: scripts/inference/inference_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    Args:
+        cfg: Dictionary of parsed cfg.yaml file.
+        img_dir: Directory of images for inference
+        batch_size (int): Number of samples per batch. Default is 8.
+
+    Returns:
+        Inference DataLoader: A PyTorch DataLoader that samples from the inference dataset.
+    """
+    
+    transform = transforms.Compose([
```
Comment (Arthurreuss) [inline scripts/inference/inference_dataloader.py:23]: we have a get_val_transforms function in transforms.py that does exactly this

File: requirements.txt
[the remark this answers is not in the document]
Comment (degrootdev) [reply requirements.txt:1]: No I forgot to add

File: scripts/inference/inference_dataloader.py (hunk 1 above)
Comment (degrootdev) [reply scripts/inference/inference_dataloader.py:23]: I think that one uses our transforms, which use a mask and a image. I use the pytorch transforms which only use a image

File: src/utils/handle_deployment.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    
+    shutil.make_archive(base_name=zip_name_output, format='zip', root_dir=cwd, base_dir=temp_output_dir)
+    shutil.rmtree(os.path.join(cwd, temp_input_dir))
+    shutil.rmtree(os.path.join(cwd, temp_output_dir))
+    
+    zip_path = os.path.join(cwd, zip_name_output + '.zip') 
+    return zip_path
+
+
+def load_model() -> torch.nn.Module:
```
[the remark this answers is not in the document]
Comment (degrootdev) [reply src/utils/handle_deployment.py:68]: Do you want this outside of this script then?

File: cfg.yaml [hunk 3]
```diff
@@ -85,5 +85,8 @@ transforms:
   - 720
   - 1280
   resize:
-  - 368 #divisible by 16, have to see how much we are downsampling
-  - 640
+  - 240 #divisible by 16, have to see how much we are downsampling
+  - 480
+inference:
+  model_path: 'outputs\baseline\2\checkpoints\chkpt_epoch_23_best.pth'
```
[the remark this answers is not in the document]
Comment (degrootdev) [reply cfg.yaml:110]: If I check eval, and see how to get best_check_point:
    
To get best_check_point, I need run_dir, to get run dir, I still need some arbitrary path which is given as an argument in eval. So I don't know if that is really better. Like I can still implement it if you want, but I do not see the benefit. as the bot need some path to some dir that is probably not the same for everyone

File: src/utils/handle_deployment.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (degrootdev) [reply src/utils/handle_deployment.py:112]: Do you just want this function on model registery and the other handle deployment functions outside of src? or only this function to model registery?
